rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,562 @@
+
+The format of a doc-comment was, however, never specified. The nix community does not understand the rules required by `nixdoc`, which we believe is one of the reasons for its lack of adoption.
+
+Here is an example of the format understood by *nixdoc*:
+
+```nix
+# nixpkgs/lib/attrsets.nix
+
+{
+  /* <Attr Description>
+  
+     Type:
+     <Type signature>
+     
+     Example:
+     <Comprehensive Code example>
+  */
+  Attr =
+    # <Describe arg1>
+    arg1:
+    # <Describe arg2>
+    arg2:
+    
+    # ... implementation
+}
+```
+
+## Current problems
+
+### Lack of having an agreed format
+
+The Nix community interprets the outer format and content of doc-comments differently. Within nixpkgs, several conventions and patterns need to be harmonized. It seems every directory decides upon their own convention; Or doesn't follow any. We immediately observe one clear thing: Most of comments utilize some form of markdown, even if they are not rendered.
+
+In general, the format for writing documentation strings is **not formally specified**.
+
+The *nixdoc*-tool requires a somewhat consistent format, but the basic format was never specified and cannot be enforced in parts of nixpkgs where nixdoc is currently not applied.
+
+Extending the scope of *nixdoc* is not the primary goal. Instead, we should find formal rules for writing *doc-comments*. Tools like *nixdoc* can then implement against this RFC instead of the format relying on nixdoc implementation details.
+
+### Impossible to differentiate from regular comments
+
+The format does not allow any distinction between doc comments and regular comments.
+
+Having a distinction would allow us to build reliable and correct documentation
+
+### References to the problems above
+
+> Note: That specific collection of links should show the amount of inconsistency within the Nix ecosystem, which is a target of this RFC.
+
+#### nixpkgs - doc-comment examples
+
+- [lib/attrsets](https://github.com/NixOS/nixpkgs/blob/master/lib/attrsets.nix)
+- [trivial-builders](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/trivial-builders.nix)
+- [stdenv/mkDerivation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix)
+- [nixos/lib/make-disk-image](https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix)
+
+#### Current tools
+
+We collected some Tools that work directly with the nix AST and comments.
+Those would benefit from a standardized doc-comment.
+
+- [noogle](https://noogle.dev) - Nix API search engine. It allows searching functions and other expressions.
+- [nix-doc](https://github.com/lf-/nix-doc) - A Nix developer tool leveraging the rnix Nix parser for intelligent documentation search and tags generation
+- [manix](https://github.com/mlvzk/manix) - A fast CLI documentation searcher for Nix.
```

Comment:
I would remove this whole section. The examples from Nixpkgs have been inserted where relevant, the mention of other tools is of unclear significance here.